processor/src/operations/ext2_ops.rs

```diff
@@ -0,0 +1,80 @@
+use super::{AdviceProvider, ExecutionError, Felt, Process};
+use vm_core::ZERO;
+
+// EXTENSION FIELD OPERATIONS
+// ================================================================================================
+
+impl<A> Process<A>
+where
+    A: AdviceProvider,
+{
+    // ARITHMETIC OPERATIONS
+    // --------------------------------------------------------------------------------------------
+    /// Gets the top four values from the stack [a1, a0, b1, b0], where a = (a1, a0) and
+    /// b = (b1, b0) are elements of the extension field, and outputs the product c = (c1, c0)
+    /// where c0 = b0 * a0 - 2 * b1 * a1 and c1 = (b0 + b1) * (a1 + a0) - b0 * a0. It pushes 0 to
+    /// the first and second positions on the stack, c1 and c2 to the third and fourth positions,
+    /// and leaves the rest of the stack unchanged.
+    pub(super) fn op_ext2mul(&mut self) -> Result<(), ExecutionError> {
+        let [b0, b1, a0, a1] = self.stack.get_top_word();
+        self.stack.set(0, ZERO);
+        self.stack.set(1, ZERO);
```

Comment:
We could do this a bit differently. Specifically, instead of reseting the first two elements to `ZERO`, we could keep original values in them. So, this would look like:
```Rust
self.stack.set(0, b0);
self.stack.set(1, b1);
```
This should allow computing inversion in 8 cycles instead of 12 like so (but double-check me on this):
```
read
read
ext2mul
movup2
eqz
assert
movup2
assert
```

Reply:
Thank you. This is great!